### PR TITLE
enh(UI): Add tile to display last time with no issue

### PR DESCRIPTION
**Pull request by Jordan91**
## Description

As a user, I want to know for a resource not OK when was the last time it was OK.

**Fixes** # (issue)

Add tile who display the  last time with no issue in detail line into detail Panel for service/host

Capture d’écran 2021-09-24 à 11 08 51

## Type of change

- [x] New functionality (non-breaking change)

## Target serie

- [x] 21.10.x (master)

<h2> How this pull request can be tested ? </h2>

Go to ressource status , select ressource to display the details panel, then you can see the new tile "Last time with no issue"

### CODE REVIEW

File: www/front_src/src/Resources/Details/tabs/Details/DetailsCard/cards.tsx [hunk 1]
```diff
@@ -121,11 +124,19 @@ const getDetailCardLines = ({
...
+        details.status.severity_code === SeverityCode.Ok,
+      line: <DetailsLine line={toDateTime(details.last_time_with_no_issue)} />,
+      title: labelLastTimeWithNoIssue,
+    },
     {
       field: details.last_check,
       line: <DetailsLine line={toDateTime(details.last_check)} />,
       title: labelLastCheck,
     },
+
```
Comment (Thebarda) [inline www/front_src/src/Resources/Details/tabs/Details/DetailsCard/cards.tsx:139]:
```suggestion
```

File: www/front_src/src/Resources/Details/tabs/Details/DetailsCard/cards.tsx [hunk 2]
```diff
@@ -121,11 +124,19 @@ const getDetailCardLines = ({
       line: <DetailsLine line={toDateTime(details.last_status_change)} />,
       title: labelLastStateChange,
     },
+    {
+      field:
+        details.last_time_with_no_issue &&
+        details.status.severity_code === SeverityCode.Ok,
```
Comment (Thebarda) [inline www/front_src/src/Resources/Details/tabs/Details/DetailsCard/cards.tsx:130]:
```suggestion
        equals(details.status.severity_code, SeverityCode.Ok),
```
You can use equals function from Ramda :)

File: www/front_src/src/Resources/Details/tabs/Details/DetailsCard/cards.tsx (hunk 1 above)
Comment (Jordan91) [reply www/front_src/src/Resources/Details/tabs/Details/DetailsCard/cards.tsx]: Thank's ;)

File: www/front_src/src/Resources/Details/tabs/Details/DetailsCard/cards.tsx (hunk 2 above)
Comment (Jordan91) [reply www/front_src/src/Resources/Details/tabs/Details/DetailsCard/cards.tsx]: Yes it's not false :) it's better by ramda

File: www/front_src/src/Resources/Details/tabs/Details/DetailsCard/cards.tsx [hunk 3]
```diff
@@ -1,6 +1,8 @@
 import * as React from 'react';
 
-import { pick } from 'ramda';
+import { equals, pick } from 'ramda';
+
+import { SeverityCode } from '@centreon/centreon-frontend/packages/centreon-ui/src';
```
Comment (bdauria) [inline www/front_src/src/Resources/Details/tabs/Details/DetailsCard/cards.tsx:5]:
```suggestion
import { SeverityCode } from '@centreon/ui';
```

**Review by Thebarda — DISMISSED** [review]

**Review by jdelpierre — DISMISSED** [review]

File: www/front_src/src/Resources/Details/tabs/Details/SortableCards/index.tsx [hunk 4]
```diff
@@ -50,7 +50,8 @@ const SortableCards = ({ panelWidth, details }: Props): JSX.Element => {
   );
 
   const displayedCards = filter(
-    ({ field }) => !isNil(field) && !isEmpty(field),
+    ({ shouldBeDisplayed }) =>
+      !isNil(shouldBeDisplayed) && !isEmpty(shouldBeDisplayed),
```
Comment (bdauria) [inline www/front_src/src/Resources/Details/tabs/Details/SortableCards/index.tsx:54]:
```suggestion
      shouldBeDisplayed,
```

File: www/front_src/src/Resources/Details/tabs/Details/DetailsCard/cards.tsx [hunk 5]
```diff
@@ -73,123 +75,133 @@ const getDetailCardLines = ({
           title={t(labelStatusInformation)}
         />
       ),
+      shouldBeDisplayed: details.information ? true : undefined,
```
Comment (bdauria) [inline www/front_src/src/Resources/Details/tabs/Details/DetailsCard/cards.tsx:78]:
```suggestion
      shouldBeDisplayed: !isNil(details.information),
```

File: www/front_src/src/Resources/Details/tabs/Details/DetailsCard/cards.tsx [hunk 6]
```diff
@@ -73,123 +75,133 @@ const getDetailCardLines = ({
...
       title: labelStatusInformation,
       xs: 12,
     },
     {
-      field: details.downtimes,
       isCustomCard: true,
       line: <DowntimesCard details={details} />,
+      shouldBeDisplayed: details.downtimes ? true : undefined,
```
Comment (bdauria) [inline www/front_src/src/Resources/Details/tabs/Details/DetailsCard/cards.tsx:85]:
```suggestion
      shouldBeDisplayed: !isNil(details.downtimes),
```

File: www/front_src/src/Resources/Details/tabs/Details/DetailsCard/cards.tsx [hunk 7]
```diff
@@ -73,123 +75,133 @@ const getDetailCardLines = ({
...
       title: labelDowntimeDuration,
       xs: 12,
     },
     {
-      field: details.acknowledgement ? true : undefined,
       isCustomCard: true,
       line: <AcknowledgementCard details={details} />,
+      shouldBeDisplayed: details.acknowledgement ? true : undefined,
```
Comment (bdauria) [inline www/front_src/src/Resources/Details/tabs/Details/DetailsCard/cards.tsx:92]:
```suggestion
      shouldBeDisplayed: !isNil(details.acknowledgement),
```

File: www/front_src/src/Resources/Details/tabs/Details/DetailsCard/cards.tsx [hunk 8]
```diff
@@ -73,123 +75,133 @@ const getDetailCardLines = ({
...
       title: labelAcknowledgement,
       xs: 12,
     },
     {
-      field: details.fqdn,
       line: <DetailsLine line={details.fqdn} />,
+      shouldBeDisplayed: details.fqdn ? true : undefined,
```
Comment (bdauria) [inline www/front_src/src/Resources/Details/tabs/Details/DetailsCard/cards.tsx:98]:
```suggestion
      shouldBeDisplayed: !isNil(details.fqdn),
```

File: www/front_src/src/Resources/Details/tabs/Details/DetailsCard/cards.tsx [hunk 9]
```diff
@@ -73,123 +75,133 @@ const getDetailCardLines = ({
...
       title: labelFqdn,
       xs: 12,
     },
     {
-      field: details.alias,
       line: <DetailsLine line={details.alias} />,
+      shouldBeDisplayed: details.alias ? true : undefined,
```
Comment (bdauria) [inline www/front_src/src/Resources/Details/tabs/Details/DetailsCard/cards.tsx:104]:
```suggestion
      shouldBeDisplayed: !isNil(details.alias),
```

File: www/front_src/src/Resources/Details/tabs/Details/DetailsCard/cards.tsx [hunk 10]
```diff
@@ -73,123 +75,133 @@ const getDetailCardLines = ({
...
       title: labelAlias,
     },
     {
-      field: details.monitoring_server_name,
       line: <DetailsLine line={details.monitoring_server_name} />,
+      shouldBeDisplayed: details.monitoring_server_name ? true : undefined,
```
Comment (bdauria) [inline www/front_src/src/Resources/Details/tabs/Details/DetailsCard/cards.tsx:109]:
```suggestion
      shouldBeDisplayed: !isNil(details.monitoring_server_name),
```

File: www/front_src/src/Resources/Details/tabs/Details/DetailsCard/cards.tsx [hunk 11]
```diff
@@ -73,123 +75,133 @@ const getDetailCardLines = ({
...
       title: labelMonitoringServer,
     },
     {
-      field: details.timezone,
       line: <DetailsLine line={details.timezone} />,
+      shouldBeDisplayed: details.timezone ? true : undefined,
```
Comment (bdauria) [inline www/front_src/src/Resources/Details/tabs/Details/DetailsCard/cards.tsx:114]:
```suggestion
      shouldBeDisplayed: !isNil(details.timezone),
```

File: www/front_src/src/Resources/Details/tabs/Details/DetailsCard/cards.tsx [hunk 12]
```diff
@@ -73,123 +75,133 @@ const getDetailCardLines = ({
...
       title: labelTimezone,
     },
     {
-      field: details.duration,
       line: <DetailsLine line={`${details.duration} - ${details.tries}`} />,
+      shouldBeDisplayed: details.duration ? true : undefined,
```
Comment (bdauria) [inline www/front_src/src/Resources/Details/tabs/Details/DetailsCard/cards.tsx:119]:
```suggestion
      shouldBeDisplayed: !isNil(details.duration),
```

File: www/front_src/src/Resources/Details/tabs/Details/DetailsCard/cards.tsx [hunk 13]
```diff
@@ -73,123 +75,133 @@ const getDetailCardLines = ({
...
       title: labelCurrentStateDuration,
     },
     {
-      field: details.last_status_change,
       line: <DetailsLine line={toDateTime(details.last_status_change)} />,
+      shouldBeDisplayed: details.last_status_change ? true : undefined,
```
Comment (bdauria) [inline www/front_src/src/Resources/Details/tabs/Details/DetailsCard/cards.tsx:124]:
```suggestion
      shouldBeDisplayed: !isNil(details.last_status_change),
```

File: www/front_src/src/Resources/Details/tabs/Details/DetailsCard/cards.tsx [hunk 14]
```diff
@@ -73,123 +75,133 @@ const getDetailCardLines = ({
...
       title: labelLastStateChange,
     },
     {
-      field: details.last_check,
+      line: <DetailsLine line={toDateTime(details.last_time_with_no_issue)} />,
+      shouldBeDisplayed:
+        details.last_time_with_no_issue &&
```
Comment (bdauria) [inline www/front_src/src/Resources/Details/tabs/Details/DetailsCard/cards.tsx:130]:
```suggestion
        !isNil(details.last_time_with_no_issue) &&
```

File: www/front_src/src/Resources/Details/tabs/Details/DetailsCard/cards.tsx [hunk 15]
```diff
@@ -73,123 +75,133 @@ const getDetailCardLines = ({
...
+        !equals(details.status.severity_code, SeverityCode.Ok)
+          ? true
+          : undefined,
+      title: labelLastCheckWithOkStatus,
+    },
+    {
       line: <DetailsLine line={toDateTime(details.last_check)} />,
+      shouldBeDisplayed: details.last_check ? true : undefined,
```
Comment (bdauria) [inline www/front_src/src/Resources/Details/tabs/Details/DetailsCard/cards.tsx:138]:
```suggestion
      shouldBeDisplayed: !isNil(details.last_check),
```

File: www/front_src/src/Resources/Details/tabs/Details/DetailsCard/cards.tsx [hunk 16]
```diff
@@ -73,123 +75,133 @@ const getDetailCardLines = ({
...
       title: labelLastCheck,
     },
     {
-      field: displayChecksIcon ? true : undefined,
       line: (
         <ChecksIcon {...pick(['active_checks', 'passive_checks'], details)} />
       ),
+      shouldBeDisplayed: displayChecksIcon ? true : undefined,
```
Comment (bdauria) [inline www/front_src/src/Resources/Details/tabs/Details/DetailsCard/cards.tsx:145]:
```suggestion
      shouldBeDisplayed: displayChecksIcon,
```

File: www/front_src/src/Resources/Details/tabs/Details/DetailsCard/cards.tsx [hunk 17]
```diff
@@ -73,123 +75,133 @@ const getDetailCardLines = ({
...
       title: labelCheck,
     },
     {
-      field: details.next_check,
       line: <DetailsLine line={toDateTime(details.next_check)} />,
+      shouldBeDisplayed: details.next_check ? true : undefined,
```
Comment (bdauria) [inline www/front_src/src/Resources/Details/tabs/Details/DetailsCard/cards.tsx:150]:
```suggestion
      shouldBeDisplayed: !isNil(details.next_check),
```

File: www/front_src/src/Resources/Details/tabs/Details/DetailsCard/cards.tsx [hunk 18]
```diff
@@ -73,123 +75,133 @@ const getDetailCardLines = ({
...
       title: labelNextCheck,
     },
     {
-      field: details.execution_time,
       line: <DetailsLine line={`${details.execution_time} s`} />,
+      shouldBeDisplayed: details.execution_time ? true : undefined,
```
Comment (bdauria) [inline www/front_src/src/Resources/Details/tabs/Details/DetailsCard/cards.tsx:155]:
```suggestion
      shouldBeDisplayed: !isNil(details.execution_time),
```

File: www/front_src/src/Resources/Details/tabs/Details/DetailsCard/cards.tsx [hunk 19]
```diff
@@ -73,123 +75,133 @@ const getDetailCardLines = ({
...
       title: labelCheckDuration,
     },
     {
-      field: details.latency,
       line: <DetailsLine line={`${details.latency} s`} />,
+      shouldBeDisplayed: details.latency ? true : undefined,
```
Comment (bdauria) [inline www/front_src/src/Resources/Details/tabs/Details/DetailsCard/cards.tsx:160]:
```suggestion
      shouldBeDisplayed: !isNil(details.latency),
```

File: www/front_src/src/Resources/Details/tabs/Details/DetailsCard/cards.tsx [hunk 20]
```diff
@@ -73,123 +75,133 @@ const getDetailCardLines = ({
...
       title: labelLatency,
     },
     {
-      field: details.percent_state_change,
       line: <PercentStateChangeCard details={details} />,
+      shouldBeDisplayed: details.percent_state_change ? true : undefined,
```
Comment (bdauria) [inline www/front_src/src/Resources/Details/tabs/Details/DetailsCard/cards.tsx:165]:
```suggestion
      shouldBeDisplayed: !isNil(details.percent_state_change),
```

File: www/front_src/src/Resources/Details/tabs/Details/DetailsCard/cards.tsx [hunk 21]
```diff
@@ -73,123 +75,133 @@ const getDetailCardLines = ({
...
       title: labelPercentStateChange,
     },
     {
-      field: details.last_notification,
       line: <DetailsLine line={toDateTime(details.last_notification)} />,
+      shouldBeDisplayed: details.last_notification ? true : undefined,
```
Comment (bdauria) [inline www/front_src/src/Resources/Details/tabs/Details/DetailsCard/cards.tsx:170]:
```suggestion
      shouldBeDisplayed: !isNil(details.last_notification),
```

File: www/front_src/src/Resources/Details/tabs/Details/DetailsCard/cards.tsx [hunk 22]
```diff
@@ -73,123 +75,133 @@ const getDetailCardLines = ({
...
       title: labelLastNotification,
     },
     {
-      field: details.notification_number,
       line: <DetailsLine line={details.notification_number.toString()} />,
+      shouldBeDisplayed: details.notification_number ? true : undefined,
```
Comment (bdauria) [inline www/front_src/src/Resources/Details/tabs/Details/DetailsCard/cards.tsx:175]:
```suggestion
      shouldBeDisplayed: !isNil(details.notification_number),
```

File: www/front_src/src/Resources/Details/tabs/Details/DetailsCard/cards.tsx [hunk 23]
```diff
@@ -73,123 +75,133 @@ const getDetailCardLines = ({
...
       title: labelCurrentNotificationNumber,
     },
     {
-      field: details.calculation_type,
       line: <DetailsLine line={details.calculation_type} />,
+      shouldBeDisplayed: details.calculation_type ? true : undefined,
```
Comment (bdauria) [inline www/front_src/src/Resources/Details/tabs/Details/DetailsCard/cards.tsx:180]:
```suggestion
      shouldBeDisplayed: !isNil(details.calculation_type),
```

File: www/front_src/src/Resources/Details/tabs/Details/DetailsCard/cards.tsx [hunk 24]
```diff
@@ -73,123 +75,133 @@ const getDetailCardLines = ({
...
       title: labelCalculationType,
     },
     {
-      field: details.groups,
       line: <Groups details={details} />,
+      shouldBeDisplayed: details.groups ? true : undefined,
```
Comment (bdauria) [inline www/front_src/src/Resources/Details/tabs/Details/DetailsCard/cards.tsx:185]:
```suggestion
      shouldBeDisplayed: !isNil(details.groups),
```

File: www/front_src/src/Resources/Details/tabs/Details/DetailsCard/cards.tsx [hunk 25]
```diff
@@ -73,123 +75,133 @@ const getDetailCardLines = ({
...
     {
-      field: details.performance_data,
       isCustomCard: true,
       line: (
         <ExpandableCard
           content={details.performance_data || ''}
           title={t(labelPerformanceData)}
         />
       ),
+      shouldBeDisplayed: details.performance_data ? true : undefined,
```
Comment (bdauria) [inline www/front_src/src/Resources/Details/tabs/Details/DetailsCard/cards.tsx:197]:
```suggestion
      shouldBeDisplayed: !isNil(details.performance_data),
```

File: www/front_src/src/Resources/Details/tabs/Details/DetailsCard/cards.tsx [hunk 26]
```diff
@@ -73,123 +75,133 @@ const getDetailCardLines = ({
...
       title: labelPerformanceData,
       xs: 12,
     },
     {
-      field: details.command_line,
       isCustomCard: true,
       line: <CommandLineCard details={details} />,
+      shouldBeDisplayed: details.command_line ? true : undefined,
```
Comment (bdauria) [inline www/front_src/src/Resources/Details/tabs/Details/DetailsCard/cards.tsx:204]:
```suggestion
      shouldBeDisplayed: !isNil(details.command_line),
```

File: www/front_src/src/Resources/Details/tabs/Details/DetailsCard/cards.tsx [hunk 27]
```diff
@@ -73,123 +75,131 @@ const getDetailCardLines = ({
...
       line: <DowntimesCard details={details} />,
+      shouldBeDisplayed: !isNil(details.downtimes),
       title: labelDowntimeDuration,
       xs: 12,
     },
     {
-      field: details.acknowledgement ? true : undefined,
       isCustomCard: true,
       line: <AcknowledgementCard details={details} />,
+      shouldBeDisplayed: !isNil(details.acknowledgement),
```
Comment (bdauria) [inline www/front_src/src/Resources/Details/tabs/Details/DetailsCard/cards.tsx:92]:
```suggestion
      shouldBeDisplayed: details.acknowledgement,
```

**Comment by Jordan91** [comment]
conflicts

**Pull request closed without merging**